aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -1,37 +1,82 @@
-/// This module provides the foundation for (collectible) Tokens often called NFTs
+/// This module provides the foundation for Tokens.
 module AptosFramework::Token {
     use Std::ASCII;
-    use Std::GUID::{Self, ID};
+    use Std::Errors;
+    use Std::Event::{Self, EventHandle};
     use Std::Option::{Self, Option};
     use Std::Signer;
+
     use AptosFramework::Table::{Self, Table};
 
-    // Error map
-    const EINSUFFICIENT_BALANCE: u64 = 0;
-    const EMISSING_CLAIMED_TOKEN: u64 = 1;
-    const EINVALID_TOKEN_MERGE: u64 = 2;
-    const EMAXIMUM_NUMBER_OF_TOKENS_FOR_COLLECTION: u64 = 3;
+    const EALREADY_HAS_BALANCE: u64 = 0;
+    const EBALANCE_NOT_PUBLISHED: u64 = 1;
+    const ECOLLECTIONS_NOT_PUBLISHED: u64 = 2;
+    const ECOLLECTION_NOT_PUBLISHED: u64 = 3;
+    const ECOLLECTION_ALREADY_EXISTS: u64 = 4;
+    const ECREATE_WOULD_EXCEED_MAXIMUM: u64 = 5;
+    const EINSUFFICIENT_BALANCE: u64 = 6;
+    const EINVALID_COLLECTION_NAME: u64 = 7;
+    const EINVALID_TOKEN_MERGE: u64 = 8;
+    const EMINT_WOULD_EXCEED_MAXIMUM: u64 = 9;
+    const ENO_BURN_CAPABILITY: u64 = 10;
+    const ENO_MINT_CAPABILITY: u64 = 11;
+    const ETOKEN_ALREADY_EXISTS: u64 = 12;
+    const ETOKEN_NOT_PUBLISHED: u64 = 13;
+    const ETOKEN_STORE_NOT_PUBLISHED: u64 = 14;
+
+    //
+    // Core data structures for holding tokens
+    //
+
+    /// Represents ownership of a the data associated with this Token
+    struct Token has store {
+        id: TokenId,
+        value: u64,
+    }
 
-    // A creator may publish multiple collections
-    struct Collections has key {
-        collections: Table<ASCII::String, Collection>,
+    /// Represents a unique identity for the token
+    struct TokenId has copy, drop, store {
+        // The creator of this token
+        creator: address,
+        // The collection or set of related tokens within the creator's account
+        collection: ASCII::String,
+        // Unique name within a collection within the creator's account
+        name: ASCII::String,
```

Comment:
Are you expecting this to be a human readable name for both collection and the NFT name?

Reply:
Yes. This part is identical to the previous token.